app/models/comment.rb

```diff
@@ -0,0 +1,13 @@
+class Comment < ApplicationRecord
+  validates :body, presence: true
+
+  has_many :comments, as: :commentable
+  belongs_to :commentable, polymorphic: true
+  
+  belongs_to :user
+  
+  def parent_post
+    return @article if defined?(@article)
+    @article = commentable.is_a?(Article) ? commentable : commentable.parent_post
+  end
```

Comment:
would it be worth caching this column on the database table for easier reference? Traversing up the comment tree like this has the potential for a Stack Overflow if the nesting gets too deep and also could lead to performance issues with enough comments being loaded onto the page.

Reply:
I used your suggestion and cached the id of the parent article to each comment